application/src/main/java/org/togetherjava/tjbot/features/help/HelpSystemHelper.java

```diff
@@ -131,6 +151,90 @@ private RestAction<Message> sendExplanationMessage(GuildMessageChannel threadCha
+     * 
+     * @param originalQuestion The first message of the thread which originates from the question
+     *        asker.
+     * @param threadChannel The thread in which the question was asked.
+     * @return An answer for the user from the AI service or a message indicating either an error or
+     *         why the message wasn't used.
+     */
+    RestAction<Message> constructChatGptAttempt(ThreadChannel threadChannel,
+            String originalQuestion) {
+        Optional<String> questionOptional = prepareChatGptQuestion(threadChannel, originalQuestion);
+        Optional<String[]> chatGPTAnswer;
+
+        String question = questionOptional.orElseThrow();
+        logger.debug("The final question sent to chatGPT: {}", question);
+        chatGPTAnswer = chatGptService.ask(question);
+
+        if (chatGPTAnswer.isEmpty()) {
+            logger.warn("Something went wrong while trying to communicate with the ChatGpt API");
+            return useChatGptFallbackMessage(threadChannel);
+        }
+
+        UnaryOperator<String> preambleResponse = """
+                Here is an AI assisted attempt to answer your question 🤖. Maybe it helps! \
+                In any case, a human is on the way 👍. To continue talking to the AI, you can use \
+                %s.
+                """::formatted;
+
+        RestAction<Message> message =
+                mentionGuildSlashCommand(threadChannel.getGuild(), ChatGptCommand.COMMAND_NAME)
+                    .map(preambleResponse)
+                    .flatMap(threadChannel::sendMessage);
+
+        for (String aiResponse : chatGPTAnswer.get()) {
+            message = message.map(aiResponse::formatted).flatMap(threadChannel::sendMessage);
+        }
+
+        return message;
+    }
+
+    private Optional<String> prepareChatGptQuestion(ThreadChannel threadChannel,
+            String originalQuestion) {
+        String questionTitle = threadChannel.getName();
+        StringBuilder questionBuilder = new StringBuilder(MAX_QUESTION_LENGTH);
+
+        if (originalQuestion.length() < MIN_QUESTION_LENGTH
+                && questionTitle.length() < MIN_QUESTION_LENGTH) {
+            return Optional.empty();
+        }
+
+        questionBuilder.append(questionTitle).append(" ");
+        if (originalQuestion.length() > MAX_QUESTION_LENGTH - questionBuilder.length()) {
+            originalQuestion =
+                    originalQuestion.substring(0, MAX_QUESTION_LENGTH - questionBuilder.length());
+        }
+
+        questionBuilder.append(originalQuestion);
+
+        StringBuilder tagBuilder = new StringBuilder();
+        int stringLength = questionBuilder.length();
+        for (ForumTag tag : threadChannel.getAppliedTags()) {
+            String tagName = tag.getName();
+            stringLength += tagName.length();
+            if (stringLength > MAX_QUESTION_LENGTH) {
+                break;
+            }
+            tagBuilder.append(String.format("%s ", tagName));
+        }
```

Comment:
i dont think this'll help, have u tested it?

Reply:
I have. All the tags of the question are appended to the beginning of the question. If you don't, there might not be enough context in the title or question to answer a question. Since we are a Java discord server, people usually don't refer directly to Java for simple questions. 

So if some asks:
```
Title: "Help" 
Question:"How do I print to the console/command line?"
```

WE all know that is meant to be Java but ChatGPT states "Sure! What language are you using?" or "Ok. Here is an example in python". But one thing people are relatively good at is clicking the right tags to append to their question (in fact isn't have at least one a requirement?).